### ISSUE TITLE
FIXED[16] backed DECIMAL(38,9) not accepted by BigQuery

### ISSUE THREAD

**Issue by ignaskukenys** [fraugster/parquet-go#61]
TLDR: `FIXED[16]` backed `DECIMAL(38, 9)` field not accepted by BigQuery: 

```
Column foo has an unexpected logical type: 0.
```

Trying to create parquet files to be loaded into BigQuery and running into issues with `fixed_len_byte_array` backed decimals.

BigQuery writes its `NUMERIC` fields to parquet as `required fixed_len_byte_array(16) foo (DECIMAL(38, 9))`:

```
$ parquet-tool meta bq_decimal.parquet
foo:          REQUIRED FIXED_LEN_BYTE_ARRAY R:0 D:0

$ parquet-tool schema bq_decimal.parquet
message schema {
  required fixed_len_byte_array(16) foo (DECIMAL(38, 9));
}
```

Recreating this schema with `fraugster/parquet-go`:

```
func TestDecimal(t *testing.T) {
	type record struct {
		Foo [16]byte `parquet:"foo"`
	}
	schema, err := parquetschema.ParseSchemaDefinition(`
		message schema {
			required fixed_len_byte_array(16) foo (DECIMAL(38, 9));
		}
	`)
	if err != nil {
		panic(fmt.Sprintf("Failed to parse schema: %v", err))
	}

	w, err := floor.NewFileWriter(
		filepath.Join(outputDir, "fraugster_decimal.parquet"),
		goparquet.WithSchemaDefinition(schema),
		goparquet.WithCompressionCodec(parquet.CompressionCodec_GZIP),
	)
	if err != nil {
		panic("Failed to create writer")
	}
	r := record{} // Note: writes all bytes as zeros which should result in zero decimal value.
	w.Write(r)
	w.Close()
}
```

The `meta` and `schema` outputs of the resulting file match the BigQuery parquet, and I can see the byte array written, however BigQuery does not accept this file:

```
$ bq load --replace --source_format=PARQUET \
    PROJECT:DATASET.fraugster_decimal fraugster_decimal.parquet
Upload complete.
Waiting on bqjob_r7d1a283391779507_0000017e2ac8cd22_1 ... (0s) Current status: DONE
BigQuery error in load operation: 
Error processing job 'PROJECT:bqjob_r7d1a283391779507_0000017e2ac8cd22_1': 
Error while reading data, error message: Column foo has an unexpected logical type: 0.
```

The issue is only reproducible with `fixed_len_byte_array` - int64 backed decimals seem to work fine (but at lower precision).

**Comment by akrennmair** [comment]
Hi @ignaskukenys, thanks for reporting this! The error message is strange, and the schema at least looks like it's the same. There's a catch though that I only stumbled upon yesterday, and that is that we were parsing `DECIMAL` only as logical types, but did not set the corresponding converted type. With no access to BigQuery, I have unfortunately no way of reproducing this, but could you try to reproduce this with the latest revision from this repository's `master` branch?

**Comment by ignaskukenys** [comment]
Thanks @akrennmair I will test it out when I get a chance.

**Comment by ignaskukenys** [comment]
Happy to confirm that in fraugster-go v0.10.0 `fixed_len_byte_array` based decimals work as expected when imported into BigQuery!

Specifically, successfully imported both:

```
message schema {
	required fixed_len_byte_array(16) foo (DECIMAL(38,9));
	required fixed_len_byte_array(32) bar (DECIMAL(76,38));
}
```

`foo` as `NUMERIC` and `bar` as `BIGNUMERIC`.
